public_asset/lib/data/network/public_asset_api.dart

```diff
@@ -0,0 +1,124 @@
+import 'package:jmap_dart_client/http/converter/identities/public_asset_identities_converter.dart';
+import 'package:jmap_dart_client/http/http_client.dart';
+import 'package:jmap_dart_client/jmap/account_id.dart';
+import 'package:jmap_dart_client/jmap/core/capability/capability_identifier.dart';
+import 'package:jmap_dart_client/jmap/core/id.dart';
+import 'package:jmap_dart_client/jmap/core/patch_object.dart';
+import 'package:jmap_dart_client/jmap/core/session/session.dart';
+import 'package:jmap_dart_client/jmap/identities/identity.dart';
+import 'package:jmap_dart_client/jmap/jmap_request.dart';
+import 'package:jmap_dart_client/jmap/mail/extensions/public_asset/get/get_public_asset_method.dart';
+import 'package:jmap_dart_client/jmap/mail/extensions/public_asset/get/get_public_asset_response.dart';
+import 'package:jmap_dart_client/jmap/mail/extensions/public_asset/public_asset.dart';
+import 'package:jmap_dart_client/jmap/mail/extensions/public_asset/set/set_public_asset_method.dart';
+import 'package:jmap_dart_client/jmap/mail/extensions/public_asset/set/set_public_asset_response.dart';
+import 'package:public_asset/domain/exceptions/public_asset_exceptions.dart';
+import 'package:uuid/uuid.dart';
+
+class PublicAssetApi {
+  const PublicAssetApi(this._httpClient, this._uuid);
+
+  final HttpClient _httpClient;
+  final Uuid _uuid;
+
+  void _checkForSupportedCapability(Session session) {
+    if (!session.capabilities.containsKey(CapabilityIdentifier.jmapPublicAsset)) {
+      throw const UnsupportedCapabilityException();
+    }
+  }
```

Comment:
Please use `isSupported` method in capability identifier extension

Reply:
We are not concerning if the request is success or not, identity is still updated regardless of the result.

Reply:
Done